android-components/components/feature/fxsuggest/src/main/java/mozilla/components/feature/fxsuggest/FxSuggestSuggestionProvider.kt

```diff
@@ -92,4 +105,21 @@ internal data class SuggestionDetails(
     val fullKeyword: String,
     val isSponsored: Boolean,
     val icon: List<UByte>?,
+    val clickInfo: FxSuggestClickInfo? = null,
 )
+
+/**
+ * Collective of fields required for fxsuggest click telemetry
+ */
+sealed interface FxSuggestClickInfo {
+    /**
+     * AMP related fields for fx suggest click telemetry
+     */
+    data class Amp(
+        val blockId: String,
+        val advertiser: String,
+        val reportingUrl: String,
```

Comment:
```suggestion
        val clickUrl: String,
```

Suggestion, minor: Let's call this field `clickUrl`, just to make it clear where it comes from.